sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JDBCRDD.scala

```diff
@@ -275,6 +275,7 @@ private[sql] class JDBCRDD(
    */
   private def compileFilter(f: Filter): String = f match {
     case EqualTo(attr, value) => s"$attr = ${compileValue(value)}"
+    case EqualNullSafe(attr, value) => s"$attr <=> ${compileValue(value)}"
```

Comment:
Since this is only in MySQL, I think the only options are:
1. don't support it
2. rewrite it as using isnull and normal equality.
3. add it to dialect.

Don't have a strong preference here. If 2 is not much slower, maybe just do 2?